packages/ember-routing/lib/system/dsl.js

```diff
@@ -26,7 +26,7 @@ class DSL {
           return true;
         }
 
-        return ['array', 'basic', 'object', 'application'].indexOf(name) === -1;
+        return ['basic', 'object', 'application'].indexOf(name) === -1;
```

Comment:
Can you also remove `object`? It was essentially the same boat as `array` and should die too...